src/main/daml/Daml/Finance/Claims/Util/Lifecycle.daml

```diff
@@ -92,6 +93,23 @@ net pending =
   in
     map createPendingNode groupedPending
 
+-- | Net pending payments on the same instrument, which also have the same tag.
+netOnTag : [Pending] -> [Pending]
+netOnTag pending =
+  let
+    groupedPending = groupOn (\e -> (e.instrument, e.tag)) pending
+    createPendingNode p =
+      let
+        aggregatedAmount = foldl (\acc e -> acc + e.amount) 0.0 p
+      in
+        Pending with
+          tag = (.tag) $ head p
```

Comment:
We are taking the head of the list three times. Although it's probably not an expensive operation, it could live in a let binding